dt-impact-tracer/src/edu/washington/cs/dt/impact/runner/Runner.java

```diff
@@ -0,0 +1,742 @@
+    protected static File origOrder = null;
+    protected static File selectionOutput1 = null;
+    protected static File selectionOutput2 = null;
+    protected static Constants.MACHINES numOfMachines = MACHINES.ONE;
+    protected static int timesToRun = 1;
+    protected static String project = "CRYSTAL";
+    protected static Constants.TEST_TYPE testType = TEST_TYPE.ORIG;
+    protected static List<String> allDTList;
+    protected static File dependentTestFile = null;
+    protected static File resolveDependences = null;
+    protected static String filesToDeleteStr = null;
+    protected static boolean getCoverage = false;
+    protected static List<String> filesToDelete = null;
+    protected static List<String> argsList = null;
+    protected static double TLGTime = 0.0;
+    protected static List<WrapperTestList> listTestList = new ArrayList<>();
+    protected static List<String> origOrderTestList = null;
+    protected static File timeOrder = null;
+    protected static int threads = 0;
+
+    protected static void parseArgs(String[] args) {
+        argsList = new ArrayList<String>(Arrays.asList(args));
+        if (argsList.contains("-help")) {
+            System.out.println("Main class that relies on program arguments to generate a regression testing "
+                    + "execution order. The following options are supported:\n"
+                    + "-technique - prioritization, selection, parallelization\n" + "-coverage - statement, function\n"
+                    + "-order - absolute, relative, random, original, time (if technique is parallelization)\n"
+                    + "-resolveDependences - when specified the output will not be affected " + "by dependent tests\n"
+                    + "-numMachines - integer value (only valid when technique is parallelization\n"
+                    + "-selectionOldVers - path to directory to older version of program's selectionOutput "
+                    + "(only valid when technique is selection)\n"
+                    + "-selectionNewVers - path to directory to newer version of program's selectionOutput "
+                    + "(only valid when technique is selection)\n"
+                    + "-origOrder - path to file containing the original order the tests are executed in\n"
+                    + "-testInputDir - path to directory to sootTestOutput\n"
+                    + "-dependentTestFile - path to file containing existing known dependent tests\n"
+                    + "-filesToDelete - path to file containing list of files to delete to clean the environment "
+                    + "each time the tests are executed\n"
+                    + "-outputFile - path to file to output the regression test order, dependent test list and "
+                    + "execution time, if unspecified the output will be sent to System.out\n"
+                    + "-help - display this message\n");
+            System.exit(0);
+        }
+        //get number of threads
```

Comment:
only this section that interprets the number of threads to use was added, nothing else was changed